tests/rptest/tests/rbac_test.py

```diff
@@ -520,3 +508,134 @@ def test_license_nag(self):
         wait_until(self._has_license_nag,
                    timeout_sec=self.LICENSE_CHECK_INTERVAL_SEC * 2,
                    err_msg="License nag failed to appear")
+
+
+class RBACEndToEndTest(RBACTestBase):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.security = SecurityConfig()
+        self.security.enable_sasl = True
+        self.security.kafka_enable_authorization = True
+        self.security.endpoint_authn_method = 'sasl'
+        self.security.require_client_auth = True
+
+        self.su_rpk = RpkTool(self.redpanda,
+                              username=self.superuser.username,
+                              password=self.superuser.password,
+                              sasl_mechanism=self.superuser.algorithm)
+        self.alice_rpk = RpkTool(self.redpanda,
+                                 username=ALICE.username,
+                                 password=ALICE.password,
+                                 sasl_mechanism=ALICE.algorithm)
+
+        self.topic0 = 'some-topic'
+        self.topic1 = 'other-topic'
+
+    def setUp(self):
+        self.redpanda.set_security_settings(self.security)
+        super().setUp()
+
+    def role_for_user(self, role: str, user: RoleMember):
+        res = self.superuser_admin.list_role_members(role=role)
+        return user in RoleMemberList.from_response(res)
+
+    def has_topics(self, client: RpkTool):
+        tps = client.list_topics()
+        return tps != [], [i for i in tps]
```

Comment:
nitpick: second return value could just be `tps` or maybe `list(tps)`

Reply:
`list(tps)` works, surprisingly